MapboxMobileEvents/MMEAPIClient.m

```diff
@@ -66,21 +77,23 @@ - (void)postEvents:(NSArray *)events completionHandler:(nullable void (^)(NSErro
     
     for (NSArray *batch in eventBatches) {
         NSURLRequest *request = [self requestForEvents:batch];
-        [self.sessionWrapper processRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
-            NSError *statusError = nil;
-            if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
-                NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
-                statusError = [self statusErrorFromRequest:request andHTTPResponse:httpResponse];
-            } else {
-                statusError = [self unexpectedResponseErrorfromRequest:request andResponse:response];
-            }
-            if (completionHandler) {
-                error = error ?: statusError;
-                completionHandler(error);
-              
-                [self.metricsManager updateMetricsFromEvents:events request:request error:error];
-            }
-        }];
+        if (request) {
+            [self.sessionWrapper processRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+                NSError *statusError = nil;
+                if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                    NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+                    statusError = [self statusErrorFromRequest:request andHTTPResponse:httpResponse];
+                } else {
+                    statusError = [self unexpectedResponseErrorfromRequest:request andResponse:response];
+                }
+                if (completionHandler) {
+                    error = error ?: statusError;
+                    completionHandler(error);
+
+                    [self.metricsManager updateMetricsFromEvents:events request:request error:error];
+                }
+            }];
+        }
```

Comment:
If request is nil (and the completion block is called for each batch) should it be called here too?

Reply:
Yes, the events/batches should be counted in a non-deferred state in this case. 👍 

I'll start a new PR for these changes. Thanks!

Reply:
np!